L1-HLT-acceptance/seeds.py

```diff
@@ -0,0 +1,136 @@
+
+if not frommenu:
+    directory = tempdir
+    utils.automate.hlt_GetConfig(directory, fnames, maxfiles, redirector)
+    files = os.listdir(directory)
+
+## Case2: *Load* the HLT menus directly from menudir
+
+if frommenu:
+    keep = True
+    directory = menudir
+    files = os.listdir(directory)
+
+## Create nested dictionary containing the HLT-L1 info for the corresponding runs
+
+runs = {}
+for file in files:
+    with open(directory + file,"r") as f:
+        data = f.read()
+
+    # Create dictionary with HLT paths as keys and hltL1 objects as values
+
+    HLT_paths={}
+    for item in data.split('\n'):
+        if "cms.Path(" in item and "process.hltL1" in item:
+            hlt = item.split("_v")[0].split("process.")[1]
+            HLT_paths[hlt] = []
+            hltl1s = item.split("process.hltL1")
+            for i in range(1,len(hltl1s)): # An HLT path can contain multiple hltL1 structures
+                hltl1 = item.split("process.hltL1")[i].split(" +")[0]
+                HLT_paths[hlt].append(hltl1)
+
+    # hltL1 combine multiple L1 seeds. Create mapping between hltl1 objects to all L1 seeds
+
+    L1_seeds={}
+    for ele in data.split("\n)\n"):
+        if " = cms.EDFilter" in ele:
+            for hltl1s in HLT_paths.values(): 
+                for hltl1 in hltl1s: # An HLT path can contain multiple hltL1 structures
+                    if hltl1 in ele and "cms.string( " in ele:
+                        L1_seeds[hltl1] = ele.split('cms.string( "')[1].split('" ),\n')[0]
+
+    # Create dictionary mapping HLT paths to L1T seeds
+
+    combined = {}
+    for key in HLT_paths.keys():
+        if key not in combined.keys():
+            combined[key] = []
+        for path in HLT_paths[key]:
+            try:
+                combined[key].append(L1_seeds[path])
+            except:
+                pass
+
+    runs[file.split(".py")[0]] = combined
+
+
+## Get all HLT paths. Mapping info in .csv file, which is then fed to TriggerAcceptance.py
+
+keys = set([hlt for run in runs for hlt in runs[run]])
+
+if not os.path.exists(savedir):
+    os.mkdir(savedir)
+
+with open(savedir+csvname,"w") as csv_file:
+    writer = csv.writer(csv_file)
+    writer.writerow(["HLT"] + runs.keys())
+
+    for key in keys:
+        # Count the max number of hltL1 structures for a single HLT path
+        ls = []
+        for run in runs.keys():
+            try:
+                ls.append(len(runs[run][key]))
+            except KeyError:
+                ls.append(len([]))
+
+        m = max(ls)
+        for i in range(m): # Iterate through all seeds in possible multiple hltL1 structures for a single HLT path
+            ls = []
+            ls.append(key)
+            for run in runs.keys():
+                try:
+                    ls.append(runs[run][key][i])
+                except KeyError:
+                    ls.append(None)
+
+            writer.writerow(ls)
+
+print("HLT_L1_seeds generated.")
```

Comment:
Suggest to extend with something like

```
print("HLT_L1_seeds generated: {}".format(savedir + csvname))
```
so that users know immediately where to look for the generated output without the need for looking at the README.

Reply:
done